pkg/operator/deploy/deploy.go

```diff
@@ -149,6 +149,40 @@ func (o *operator) resources() ([]runtime.Object, error) {
 		domain += "." + o.env.Domain()
 	}
 
+	cluster := &arov1alpha1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: arov1alpha1.SingletonClusterName,
+		},
+		Spec: arov1alpha1.ClusterSpec{
+			ResourceID:    o.oc.ID,
+			Domain:        domain,
+			ACRDomain:     o.env.ACRDomain(),
+			AZEnvironment: o.env.Environment().Name,
+			Location:      o.env.Location(),
+			VnetID:        vnetID,
+			GenevaLogging: arov1alpha1.GenevaLoggingSpec{
+				ConfigVersion:            o.env.ClusterGenevaLoggingConfigVersion(),
+				MonitoringGCSEnvironment: o.env.ClusterGenevaLoggingEnvironment(),
+			},
+			APIIntIP:                 o.oc.Properties.APIServerProfile.IntIP,
+			IngressIP:                o.oc.Properties.IngressProfiles[0].IP,
+			GatewayPrivateEndpointIP: o.oc.Properties.NetworkProfile.GatewayPrivateEndpointIP,
+		},
+	}
+
+	if o.oc.Properties.NetworkProfile.GatewayPrivateEndpointIP != "" {
+		cluster.Spec.GatewayDomains = o.env.GatewayDomains()
```

Comment:
> do we need `cluster.Spec.InternetChecker = arov1alpha1.InternetCheckerSpec{}` for the upgrade case?

@mjudeikis @jewzaam answer to this question was **no**: this function is creating a brand new desired state, not transforming an existing object.  Since the fixes in #1340, when the new desired state overwrites the existing object, it plasters over existing values.